genai_stack/memory/langchain.py

```diff
@@ -0,0 +1,30 @@
+from langchain.memory import ConversationBufferMemory as cbm
+from .base import BaseMemory
+
+class ConversationBufferMemory(BaseMemory):
+    memory = None
+    input_text = None
+    output_text = None
+
+    def _post_init(self, *args, **kwargs):
+        self.memory = cbm(return_messages=True)
+
+    def add_text(self, user_text, model_text):
+        self.input_text = user_text
+        self.output_text = model_text
+        self.memory.save_context({"input": user_text}, {"output": model_text})
+
+    def get_user_text(self):
+        return self.input_text
```

Comment:
This is wrong actually because everytime you store the memory it should go to the storage and everytime you retrieve it should retrieve from the storage only you shouldn't store the values in variables since they would not be persistent.